src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/CopyAatfControllerTests.cs

```diff
@@ -0,0 +1,311 @@
+    using FakeItEasy;
+    using FluentAssertions;
+    using Web.Infrastructure;
+    using Xunit;
+
+    public class CopyAatfControllerTests
+    {
+        private readonly Fixture fixture;
+        private readonly IMapper mapper;
+        private readonly IWeeeClient weeeClient;
+        private readonly IList<CountryData> countries;
+        private readonly BreadcrumbService breadcrumbService;
+        private readonly IWeeeCache cache;
+        private readonly CopyAatfController controller;
+
+        public CopyAatfControllerTests()
+        {
+            fixture = new Fixture();
+            mapper = A.Fake<IMapper>();
+            weeeClient = A.Fake<IWeeeClient>();
+            countries = A.Dummy<IList<CountryData>>();
+            breadcrumbService = A.Fake<BreadcrumbService>();
+            cache = A.Fake<IWeeeCache>();
+
+            controller = new CopyAatfController(() => weeeClient, breadcrumbService, mapper, cache);
+        }
+
+        [Fact]
+        public void ControllerMustHaveAuthorizeClaimsAttribute()
+        {
+            typeof(CopyAatfController).Should().BeDecoratedWith<AuthorizeInternalClaimsAttribute>(a => a.Match(new AuthorizeInternalClaimsAttribute(Claims.InternalAdmin)));
+        }
+
+        [Theory]
+        [InlineData(FacilityType.Aatf, "Copy AATF for new compliance year")]
+        public async Task CopyGet_CanEdit_SetsInternalBreadcrumb(FacilityType facilityType, string expectedBreadcrumb)
+        {
+            var aatf = fixture.Build<AatfData>().With(a => a.CanEdit, true).Create(); 
+            aatf.FacilityType = facilityType;
+
+            var aatfViewModel = fixture.Create<CopyAatfViewModel>();
+
+            A.CallTo(() => weeeClient.SendAsync(A<string>._, A<GetAatfById>._)).Returns(aatf);
+
+            A.CallTo(() => mapper.Map<CopyAatfViewModel>(aatf)).Returns(aatfViewModel);
+
+            await controller.CopyAatfDetails(aatf.Id);
+
+            Assert.Equal(expectedBreadcrumb, breadcrumbService.InternalActivity);
+            A.CallTo(() => weeeClient.SendAsync(A<string>._, A<GetAatfById>._)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => mapper.Map<CopyAatfViewModel>(aatf)).MustHaveHappenedOnceExactly();
+        }
+
+        [Theory]
+        [InlineData(FacilityType.Ae, "Copy AE for new compliance year")]
+        public async Task CopyGetAe_CanEdit_SetsInternalBreadcrumb(FacilityType facilityType, string expectedBreadcrumb)
+        {
+            var aatf = fixture.Build<AatfData>().With(a => a.CanEdit, true).Create();
+            aatf.FacilityType = facilityType;
+
+            var aatfViewModel = fixture.Create<CopyAeViewModel>();
+
+            A.CallTo(() => weeeClient.SendAsync(A<string>._, A<GetAatfById>._)).Returns(aatf);
+
+            A.CallTo(() => mapper.Map<CopyAeViewModel>(aatf)).Returns(aatfViewModel);
+
+            await controller.CopyAatfDetails(aatf.Id);
+
+            Assert.Equal(expectedBreadcrumb, breadcrumbService.InternalActivity);
```

Comment:
Dont bother with this and the below check if this is just testing the breadcrumb